lib/get-open-port.js

```diff
@@ -0,0 +1,20 @@
+const getPort = require('get-port')
+/*
+There is a very tiny chance of a race condition if another process starts using the same port number as you in between the time you get the port number and you actually start using it.
+
+Race conditions in the same process are mitigated against by using a lightweight locking mechanism where a port will be held for a minimum of 15 seconds before being released again.
+*/
+const lockedPorts = new Map()
+
+function getOpenPort (cb) {
+  getPort().then((port) => {
+    if (lockedPorts.get(port)) getOpenPort(null, cb)
+    lockedPorts.set(port, true)
+    setTimeout(() => {
+      lockedPorts.set(port, false)
```

Comment:
If multiple clients sync one after the other (so the sync process stays up) then this could cancel a sync. We should work in the sync code to gracefully restart broken connections anyway, so that this is not an issue.

Reply:
This is just to find an open port so no processes will be cancelled as a result of this function

Reply:
Sorry yes my bad. The issue is with the port not being locked after timeout, so the (small) possibility of another process trying to use it.

Reply:
The port is locked after timeout if it's taken by a process, since get port will only return a currently open port

Reply:
Got it, sorry was being dumb, I had mis-read your comment!